oozie-to-airflow/mappers/spark_mapper.py

```diff
@@ -45,190 +59,171 @@ def __init__(
         self.template = template
-        self.params = params
+        self.params = params or {}
         self.trigger_rule = trigger_rule
-        self._parse_oozie_node(oozie_node)
-
-    def _parse_oozie_node(self, oozie_node: ET.Element):
-        """
-        Property values specified in the configuration element override
-        values specified in the job-xml file.
-        """
-        self.application = ""
-        self.conf = {}
-        self.conn_id = "spark_default"
-        self.files = None
-        self.py_files = None
-        self.driver_classpath = None
-        self.jars = None
-        self.java_class = None
-        self.packages = None
-        self.exclude_packages = None
-        self.repositories = None
-        self.total_executor_cores = None
-        self.executor_cores = None
-        self.executor_memory = None
-        self.driver_memory = None
-        self.keytab = None
-        self.principal = None
-        self.spark_name = "airflow-spark"
-        self.num_executors = None
+        self.java_class = ""
+        self.java_jar = ""
+        self.job_name = None
+        self.jars = []
+        self.properties = {}
         self.application_args = []
-        self.env_vars = None
-        self.verbose = False
+        self.file_extractor = FileExtractor(oozie_node=oozie_node, params=self.params)
+        self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, params=self.params)
+        self.prepare_command = None
+        self.hdfs_files = []
+        self.hdfs_archives = []
+        self.dataproc_jars = []
 
-        # Prepare nodes
-        self.delete_paths = []
-        self.mkdir_paths = []
+    def on_parse_node(self):
 
-        prepare_nodes = xml_utils.find_nodes_by_tag(oozie_node, "prepare")
+        if self.has_prepare:
+            self.prepare_command = self.get_prepare_command(oozie_node=self.oozie_node, params=self.params)
 
-        if prepare_nodes:
-            # If there exists a prepare node, there will only be one, according
-            # to oozie xml schema
-            self.delete_paths, self.mkdir_paths = self.parse_prepare_node(prepare_nodes[0])
+        _, self.hdfs_files = self.file_extractor.parse_node()
+        _, self.hdfs_archives = self.archive_extractor.parse_node()
 
-        # master url, deploy mode,
-        self.application = self.test_and_set(oozie_node, "jar", "''", params=self.params, quote=True)
-        self.spark_name = self.test_and_set(
-            oozie_node, "name", "'airflow-spark'", params=self.params, quote=True
-        )
-        self.java_class = self.test_and_set(oozie_node, "class", None, params=self.params, quote=True)
+        self.java_jar = self._get_or_default(self.oozie_node, SPARK_TAG_JAR, None, params=self.params)
+        self.java_class = self._get_or_default(self.oozie_node, SPARK_TAG_CLASS, None, params=self.params)
+        if self.java_class and self.java_jar:
+            self.dataproc_jars = [self.java_jar]
```

Comment:
It's worth noting that according to the docs:
> The jar element indicates a comma separated list of jars or python files.

So I see 2 problems here:
1. In this case this may be a one-element list, with the element being a comma-separated list in a string ;)
2. These can be Python files, which might be rejected by the Dataproc API
  - With regards to 2) we could probably write in the docs that we don't support Python files here.

Reply:
Can you provide a link to which documentation you are looking at?

Reply:
Sorry, my bad: https://oozie.apache.org/docs/5.1.0/DG_SparkActionExtension.html

![image](https://user-images.githubusercontent.com/4280682/56965079-b89d7180-6b5c-11e9-9f6e-6d43bf0165a7.png)

Reply:
We can skip this for now. Maybe let's just mention in the readme that we don't support this yet.

Reply:
Yeah. We mentioned it previously in the design as far as I remember.

Reply:
I added a note in README.md